Utilities/XrdAdaptor/src/QualityMetric.cc

```diff
@@ -157,16 +157,17 @@ QualityMetricFactory * QualityMetricFactory::m_instance = new QualityMetricFacto
 std::unique_ptr<QualityMetricSource>
 QualityMetricFactory::get(timespec now, const std::string &id)
```

Comment:
Ah crap, the "real" initialization of `m_id` got rearranged in a previous update (when we added support for site names).  It should be the data server instead (still possible to get two concurrent requests though!).

By "race condition", are you referring to before or after the proposed changes?